src/tpm.rs

```diff
@@ -29,30 +29,510 @@ const RETRY: usize = 4;
+            common::BOOTSTRAP_KEY_SIZE,
+            owner_pw
+        ),
+        None,
+    )?;
+
+    if ret_out.len() as i32 == common::BOOTSTRAP_KEY_SIZE {
+        return Ok(ret_out.into());
+    }
+    Ok(String::new())
+}
+
+/*
+ *  Input:
+ *      key to write to nvram
+ *  Return:
+ *      Error if process failed
+ *
+ *  Write data to a Non-Volatile (NV) index.
+ */
+fn tpm_write_key_nvram(key: String) -> Result<(), KeylimeTpmError> {
+    let owner_pw = get_tpm_metadata_content("owner_pw")?;
+    let mut tf = NamedTempFile::new()?;
+    tf.write_all(key.as_bytes())?;
+    let attrs = "ownerread|policywrite|ownerwrite";
+    let tf_path = temp_file_get_path(&tf)?;
+
+    run(format!("tpm2_nvdefine -x 0x1500018 -a 0x40000001 -s {} -t \"{}\" -p {} -P {}", common::BOOTSTRAP_KEY_SIZE, attrs, owner_pw, owner_pw), None)?;
+    run(
+        format!(
+            "tpm2_nvwrite -x 0x1500018 -a 0x40000001 -P {} {}",
+            owner_pw, tf_path
+        ),
+        None,
+    )?;
+    Ok(())
+}
+
+/*
+ *  Input:
+ *      activation flag
+ *      asymmetric algorithm
+ *      hash algorithm
+ *      sign algorithm
+ *  Return:
+ *      Error if the process failed.
+ *
+ *  Create a new aik key from TPM. Store the aik_name, aik_password,
+ *  aik_public_key and aik_handle to tpmdata.json file.
+ */
+fn tpm_create_aik(
+    activate: bool,
+    asym_alg: String,
+    hash_alg: String,
+    sign_alg: String,
+) -> Result<(), KeylimeTpmError> {
+    let owner_pw = get_tpm_metadata_content("owner_pw")?;
+    let aik = get_tpm_metadata_content("aik")?;
+    let aik_name = get_tpm_metadata_content("aik_name")?;
+
+    if !aik.is_empty() && !aik_name.is_empty() {
+        let aik_handle = get_tpm_metadata_content("aik_handle")?;
+
+        let (ret_out, _) =
+            run("tpm2_getcap -c handles-persisitent".to_string(), None)?;
+        let ret_out_map: Value = serde_yaml::from_str(&ret_out)?;
+        if ret_out_map[&aik_handle].is_null() {
+            run(
+                format!(
+                    "tpm2_evictcontrol -a o -c {} -P {}",
+                    aik_handle, owner_pw
+                ),
+                None,
+            )?;
+        }
+        set_tpm_metadata_content("aik", "")?;
+        set_tpm_metadata_content("aik_name", "")?;
+        set_tpm_metadata_content("aik_pw", "")?;
+        set_tpm_metadata_content("aik_handle", "")?;
+    }
+
+    let ek_handle = get_tpm_metadata_content("ek_handle")?;
+    if ek_handle.is_empty() {
+        return Err(KeylimeTpmError::new_tpm_rust_error(
+            "create ek failed. ek_handle is missing",
+        ));
+    }
+    let aik_pw = random_password(20)?;
+
+    let tf = NamedTempFile::new()?;
+    let aik_pub_path = temp_file_get_path(&tf)?;
+    let paths = vec![aik_pub_path];
+    let (ret_out, f_out) = run(
+        format!(
+            "tpm2_createak -C {} -k - -G {} -D {} -s {} -p {} -f pem -e {} -P {} -o {}",
```

Comment:
another legacy tools check need (I know its a PIA, but we will need this for RHEL 7 support);

https://github.com/keylime/keylime/blob/94328329d2dad5d5a659168865d2e73347a48ce4/keylime/tpm2.py#L551-L554